recipes/abseil/all/conanfile.py

```diff
@@ -78,32 +78,41 @@ def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
         cmake = self._configure_cmake()
-        abi_h = tools.load("abi.h")
-        abi = dict()
-        for line in abi_h.splitlines():
-            if line.startswith("#define"):
-                tokens = line.split()
-                if len(tokens) == 3:
-                    abi[tokens[1]] = tokens[2]
-        for name, value in abi.items():
-            tools.replace_in_file(os.path.join(self._source_subfolder, "absl", "base", "options.h"),
-                    "#define ABSL_OPTION_{} 2".format(name),
-                    "#define ABSL_OPTION_{} {}".format(name, value))
+        abi_file = _ABIFile("abi.h")
+        abi_file.replace_in_options_file(os.path.join(self._source_subfolder, "absl", "base", "options.h"))
         cmake.build()
 
+    @property
+    def _module_path(self):
+        return os.path.join("lib", "cmake", "conan_trick")
+
+    @property
+    def _cxx_std_build_module(self):
+        return "cxx_std.cmake"
+
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         cmake_folder = os.path.join(self.package_folder, "lib", "cmake")
-        self._create_components_file_from_cmake_target_file(os.path.join(cmake_folder, "absl", "abslTargets.cmake"))
+        components = self._create_components_file_from_cmake_target_file(os.path.join(cmake_folder, "absl", "abslTargets.cmake"))
         tools.rmdir(cmake_folder)
 
+        # Create a build-module that will propagate the required cxx_std to consumers of this recipe's targets
+        # TODO: Revisit with feedback from https://github.com/conan-io/conan/issues/10281
+        os.makedirs(os.path.join(self.package_folder, self._module_path))
+        with open(os.path.join(self.package_folder, self._module_path, self._cxx_std_build_module), 'w', encoding='utf-8') as f:
```

Comment:
strictly speaking, should be no longer necessary for 3.7+ https://docs.python.org/3/library/os.html#utf8-mode

Reply:
uhm... the linter runs with python 3.8 and it shows a warning: https://github.com/conan-io/conan-center-index/blob/47d937cab2348946b7956ceaea5e10502ddf2a98/.github/workflows/linters.yml#L21

Reply:
@ericLemanissier is it a false positivie of the linter?
```
[open()](https://docs.python.org/3/library/functions.html#open), io.open(), and [codecs.open()](https://docs.python.org/3/library/codecs.html#codecs.open) use the UTF-8 encoding by default.
```